test_plans/sycl_external.asciidoc

```diff
@@ -0,0 +1,64 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for device image
+
+This is a test plan for SYCL_EXTERNAL macro as described in Section 5.10. of the SYCL 2020 specification.
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are not backend-specific and are performed for any SYCL backend.
+
+=== Device coverage
+
+All tests construct a test device for which conformance is assessed. All the tests described below are performed once for that test device.
+
+== Tests
+
+All test cases a function is declared with SYCL_EXTERNAL in translation unit with test case and defined in some translation unit, where the function is also declared with SYCL_EXTERNAL.
+This function is called in test case from both host and device code. From device code it's called from both single_task and parallel_for (for 1 item).
+Function has accessor as a parameter and is used to assign value to it. For call from host host_accessor is used.
```

Comment:
Note that `SYCL_EXTERNAL` is an optional feature, so implementations are not required to support it.  I think this test needs to use `#ifdef SYCL_EXTERNAL`, and the test should be skipped if the implementation doesn't define that macro.

Reply:
Added condition for `#ifdef SYCL_EXTERNAL`.